### PR TITLE
Perf script improvements

### PR DESCRIPTION
**Pull request by jtattermusch**
-- support overriding client language in a scenario (mostly useful for running C++ client to test throughput of servers in different languages).
-- use different sized of DEEP and WIDE for sync and async clients

-- change default category for run_performance_tests.py to 'all' as we are already setting the smoketest category properly in https://github.com/grpc/grpc/blob/master/tools/jenkins/run_performance.sh

### CODE REVIEW

File: tools/run_tests/performance/scenario_config.py [hunk 1]
```diff
@@ -560,25 +586,28 @@ def scenarios(self):
           secure=secure,
           categories=smoketest_categories)
 
+      # unconstrained_client='async' is intended (client uses goroutines)
       yield _ping_pong_scenario(
           'go_protobuf_sync_unary_qps_unconstrained_%s' % secstr, rpc_type='UNARY',
           client_type='SYNC_CLIENT', server_type='SYNC_SERVER',
-          use_unconstrained_client=True,
+          unconstrained_client='async',
```
Comment (vjpai) [inline tools/run_tests/performance/scenario_config.py]: sync?

File: tools/run_tests/performance/scenario_config.py [hunk 2]
```diff
@@ -560,25 +586,28 @@ def scenarios(self):
...
           secure=secure,
           categories=smoketest_categories)
 
+      # unconstrained_client='async' is intended (client uses goroutines)
       yield _ping_pong_scenario(
           'go_protobuf_sync_streaming_qps_unconstrained_%s' % secstr, rpc_type='STREAMING',
           client_type='SYNC_CLIENT', server_type='SYNC_SERVER',
-          use_unconstrained_client=True,
+          unconstrained_client='async',
```
Comment (vjpai) [inline tools/run_tests/performance/scenario_config.py]: sync?

File: tools/run_tests/performance/scenario_config.py [hunk 3]
```diff
@@ -560,25 +586,28 @@ def scenarios(self):
...
           secure=secure)
 
+      # unconstrained_client='async' is intended (client uses goroutines)
       # ASYNC_GENERIC_SERVER for Go actually uses a sync streaming server,
       # but that's mostly because of lack of better name of the enum value.
       yield _ping_pong_scenario(
           'go_generic_sync_streaming_qps_unconstrained_%s' % secstr, rpc_type='STREAMING',
           client_type='SYNC_CLIENT', server_type='ASYNC_GENERIC_SERVER',
-          use_unconstrained_client=True, use_generic_payload=True,
+          unconstrained_client='async', use_generic_payload=True,
```
Comment (vjpai) [inline tools/run_tests/performance/scenario_config.py]: sync?

**Comment by vjpai** [comment]
Maybe for go the async values are appropriate since it uses go-routines rather than true system-level threads?

**Comment by vjpai** [comment]
LGTM except for the Go sync/async issue

File: tools/run_tests/performance/scenario_config.py (hunk 1 above)
Comment (jtattermusch) [inline tools/run_tests/performance/scenario_config.py]: Yes, go uses goroutines and not real threads, so using the async values for unconstrained seems to be fine (also adding a comment a few lines above).

I am planning to leave this as 'async' unless there's a good reason not to.

**Comment by vjpai** [comment]
LGTM.

**Pull request merged**